daemon/networkdriver/bridge/driver.go

```diff
@@ -614,27 +599,21 @@ func Release(job *engine.Job) error {
 			logrus.Infof("Unable to release IPv6 %s", err)
 		}
 	}
-	return nil
 }
 
 // Allocate an external port and map it to the interface
-func AllocatePort(job *engine.Job) error {
+func AllocatePort(id string, port nat.Port, binding nat.PortBinding) (nat.PortBinding, error) {
 	var (
-		err error
-
 		ip            = defaultBindingIP
-		id            = job.Args[0]
-		hostIP        = job.Getenv("HostIP")
-		hostPort      = job.GetenvInt("HostPort")
-		containerPort = job.GetenvInt("ContainerPort")
-		proto         = job.Getenv("Proto")
+		proto         = port.Proto()
+		containerPort = port.Int()
```

Comment:
@LK4D4 at least I don't think I'm modifying any behavior in this PR. Actually the only modification I'm making is that if somebody types in a negative port, then this PR will error out, while master would just consider it 0 (to be verified but that's what i see from the code)